api_tests/registrations/views/test_registration_forks.py

```diff
@@ -19,269 +16,327 @@
+        assert data['attributes']['registration'] == False
+        assert data['attributes']['fork'] == True
+
+    def test_create_fork_from_private_registration_with_new_title(self, app, user, private_registration, private_registration_url, fork_data_with_title):
+        res = app.post_json_api(private_registration_url, fork_data_with_title, auth=user.auth)
+        assert res.status_code == 201
         data = res.json['data']
-        assert_equal(data['id'], self.private_registration.forks.first()._id)
-        assert_equal(data['attributes']['title'], self.fork_data_with_title['data']['attributes']['title'])
-        assert_equal(data['attributes']['registration'], False)
-        assert_equal(data['attributes']['fork'], True)
-
-    def test_can_fork_public_registration_logged_in(self):
-        res = self.app.post_json_api(self.public_registration_url, self.fork_data, auth=self.user_two.auth)
-        assert_equal(res.status_code, 201)
+        assert data['id'] == private_registration.forks.first()._id
+        assert data['attributes']['title'] == fork_data_with_title['data']['attributes']['title']
+        assert data['attributes']['registration'] == False
+        assert data['attributes']['fork'] == True
+
+    def test_can_fork_public_registration_logged_in(self, app, user_two, public_registration, public_registration_url, fork_data):
+        res = app.post_json_api(public_registration_url, fork_data, auth=user_two.auth)
+        assert res.status_code == 201
         data = res.json['data']
-        assert_equal(data['id'], self.public_registration.forks.first()._id)
-        assert_equal(data['attributes']['title'], 'Fork of ' + self.public_registration.title)
-        assert_equal(data['attributes']['registration'], False)
-        assert_equal(data['attributes']['fork'], True)
-
-    def test_cannot_fork_public_registration_logged_out(self):
-        res = self.app.post_json_api(self.public_registration_url, self.fork_data, expect_errors=True)
-        assert_equal(res.status_code, 401)
-        assert_equal(res.json['errors'][0]['detail'], 'Authentication credentials were not provided.')
-
-    def test_can_fork_public_registration_logged_in_contributor(self):
-        res = self.app.post_json_api(self.public_registration_url, self.fork_data, auth=self.user.auth)
-        assert_equal(res.status_code, 201)
+        assert data['id'] == public_registration.forks.first()._id
+        assert data['attributes']['title'] == 'Fork of ' + public_registration.title
+        assert data['attributes']['registration'] == False
+        assert data['attributes']['fork'] == True
+
+    def test_cannot_fork_public_registration_logged_out(self, app, public_registration_url, fork_data):
+        res = app.post_json_api(public_registration_url, fork_data, expect_errors=True)
+        assert res.status_code == 401
+        assert res.json['errors'][0]['detail'] == 'Authentication credentials were not provided.'
```

Comment:
^ might be outside the scope of this PR, though. This change can be made in a follow-up PR.